src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.cs

```diff
@@ -1118,12 +1109,10 @@ public SolutionCompilationState WithoutFrozenSourceGeneratedDocuments()
                     }
 
                     trackerMap[projectId] = new GeneratedFileReplacingCompilationTracker(existingTracker, new(documentStatesForProject));
-                    mapModified = true;
                 }
-
-                return mapModified;
             },
-            (documentStatesByProjectId, this.SolutionState));
+            (documentStatesByProjectId, this.SolutionState),
+            skipEmptyCallback: documentStatesByProjectId.Count == 0);
```

Comment:
This is never empty here due to the following:
https://github.com/dotnet/roslyn/blob/6f57c67c64daab8ecaa0e8e45365482a630a5543/src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.cs#L1091-L1093

```suggestion
            skipEmptyCallback: false);
```